lxd/auth/oidc/oidc.go

```diff
@@ -19,32 +22,175 @@ type Verifier struct {
 	keySet   oidc.KeySet
 	verifier op.AccessTokenVerifier
 
-	clientID string
-	issuer   string
-	audience string
+	clientID  string
+	issuer    string
+	audience  string
+	cookieKey []byte
 }
 
 // Auth extracts the token, validates it and returns the user information.
-func (o *Verifier) Auth(ctx context.Context, r *http.Request) (string, error) {
-	// When a client wants to authenticate, it needs to set the Authorization HTTP header like this:
-	//    Authorization Bearer <access_token>
-	// If set correctly, LXD will attempt to verify the access token, and grant access if it's valid.
-	// If the verification fails, LXD will return an InvalidToken error. The client should then either use its refresh token to get a new valid access token, or log in again.
-	// If the Authorization header is missing, LXD returns an AuthenticationRequired error.
-	// Both returned errors contain information which are needed for the client to authenticate.
-	parts := strings.Split(r.Header.Get("Authorization"), "Bearer ")
-	if len(parts) != 2 {
-		return "", fmt.Errorf("Bad authorization token, expected a Bearer token")
+func (o *Verifier) Auth(ctx context.Context, w http.ResponseWriter, r *http.Request) (string, error) {
+	var token string
+
+	auth := r.Header.Get("Authorization")
+	if auth != "" {
+		// When a client wants to authenticate, it needs to set the Authorization HTTP header like this:
+		//    Authorization Bearer <access_token>
+		// If set correctly, LXD will attempt to verify the access token, and grant access if it's valid.
+		// If the verification fails, LXD will return an InvalidToken error. The client should then either use its refresh token to get a new valid access token, or log in again.
+		// If the Authorization header is missing, LXD returns an AuthenticationRequired error.
+		// Both returned errors contain information which are needed for the client to authenticate.
+		parts := strings.Split(auth, "Bearer ")
```

Comment:
Just an FYI, not a blocker, but there is `strings.Cut()` now which would be quite good in this situation.

Reply:
Ah, good to know. It's not new code in this case but maybe something we can apply at large eventually.